src/herbpy/herbrobot.py

```diff
@@ -30,6 +34,32 @@ def __init__(self, left_arm_sim, right_arm_sim, right_ft_sim,
 
         Robot.__init__(self, robot_name='herb')
 
+
+        # Controller setup
+        # TODO load controller names from same config as ros_control?
+        self.controller_manager = None  # TODO verify never used in sim
+        self.controllers_always_on = []
+        controllers_manip = []
+
+        # TODO how to deal with sim on controller manager?
+        self.full_controller_sim = (left_arm_sim and right_arm_sim and
+                                    left_ft_sim and right_ft_sim and
+                                    left_hand_sim and right_hand_sim and
+                                    head_sim)
+        if not self.full_controller_sim:
+            # any non-simulation requires ros and the ros_control stack
+            import rospy
+            if not rospy.core.is_initialized():
+                rospy.init_node('herbpy')
```

Comment:
I prefer to only call `init_node` in the top-level script that serves as the entry point for the application. This function may only be called once per process and takes several options that we cannot pass here.